src/qt/askpassphrasedialog.cpp

```diff
@@ -216,18 +218,56 @@ void AskPassphraseDialog::textChanged()
     {
     case Encrypt: // New passphrase x2
         acceptable = !ui->passEdit2->text().isEmpty() && !ui->passEdit3->text().isEmpty();
+        this->updatePasswordStrengthIndicator(ui->passEdit2);
         break;
     case Unlock: // Old passphrase x1
     case Decrypt:
         acceptable = !ui->passEdit1->text().isEmpty();
         break;
     case ChangePass: // Old passphrase x1, new passphrase x2
         acceptable = !ui->passEdit1->text().isEmpty() && !ui->passEdit2->text().isEmpty() && !ui->passEdit3->text().isEmpty();
+        this->updatePasswordStrengthIndicator(ui->passEdit2);
         break;
     }
     ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(acceptable);
 }
 
+void AskPassphraseDialog::updatePasswordStrengthIndicator(QLineEdit *lineedit)
+{
+    std::string pass = lineedit->text().toUtf8().constData();
+    bool mixedcase = (boost::to_upper_copy<std::string>(pass) != pass) && (boost::to_lower_copy<std::string>(pass) != pass);
+    bool specialChars = pass.find_first_not_of("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxzy1234567890") != std::string::npos;
```

Comment:
should we add digit checks too? Seems like right now we only consider lowercase, uppercase, special characters and length.

Reply:
Thanks will add this later